fuse/ipns/ipns_unix.go

```diff
@@ -211,91 +217,85 @@ type Node struct {
 
 	Ipfs   *core.IpfsNode
 	Nd     *mdag.Node
-	dagMod *uio.DagModifier
+	dagMod *mod.DagModifier
 	cached *ftpb.Data
 }
+*/
 
+type Directory struct {
+	dir *nsfs.Directory
+}
+
+type File struct {
+	fi nsfs.File
+
+	fs.NodeRef
+}
+
+/*
 func (s *Node) loadData() error {
 	s.cached = new(ftpb.Data)
 	return proto.Unmarshal(s.Nd.Data, s.cached)
 }
+*/
 
 // Attr returns the attributes of a given node.
-func (s *Node) Attr() fuse.Attr {
-	if s.cached == nil {
-		err := s.loadData()
-		if err != nil {
-			log.Debugf("Error loading PBData for file: '%s'", s.name)
-		}
-	}
-	switch s.cached.GetType() {
-	case ftpb.Data_Directory:
-		return fuse.Attr{Mode: os.ModeDir | 0555}
-	case ftpb.Data_File, ftpb.Data_Raw:
-		size, err := ft.DataSize(s.Nd.Data)
-		if err != nil {
-			log.Debugf("Error getting size of file: %s", err)
-			size = 0
-		}
-		if size == 0 {
-			size = s.dagMod.Size()
-		}
-
-		mode := os.FileMode(0666)
-		if IpnsReadonly {
-			mode = 0444
-		}
+func (d *Directory) Attr() fuse.Attr {
+	log.Debug("Directory Attr")
+	return fuse.Attr{Mode: os.ModeDir | 0555}
+}
 
-		return fuse.Attr{
-			Mode:   mode,
-			Size:   size,
-			Blocks: uint64(len(s.Nd.Links)),
-		}
-	default:
-		log.Debug("Invalid data type.")
-		return fuse.Attr{}
+// Attr returns the attributes of a given node.
+func (fi *File) Attr() fuse.Attr {
+	log.Debug("File Attr")
+	size, err := fi.fi.Size()
+	if err != nil {
+		log.Critical("Failed to get file size: %s", err)
+	}
+	return fuse.Attr{
+		Mode: os.FileMode(0666),
+		Size: uint64(size),
```

Comment:
Yeah, it doesnt really mean anything. All it does (as far as i know) is fill out a field in the stat output. And we dont have an easy way of counting blocks